streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableImpl.java

```diff
@@ -130,34 +130,45 @@ public String queryableStoreName() {
         // only materialize if the state store is queryable
         final boolean shouldMaterialize = materializedInternal != null && materializedInternal.isQueryable();
 
-        KTableProcessorSupplier<K, V, V> processorSupplier = new KTableFilter<>(this,
-                predicate,
-                filterNot,
-                shouldMaterialize ? materializedInternal.storeName() : null);
+        final KTableProcessorSupplier<K, V, V> processorSupplier = new KTableFilter<>(
+            this,
+            predicate,
+            filterNot,
+            shouldMaterialize ? materializedInternal.storeName() : null
+        );
+
+        final ProcessorParameters<K, V> processorParameters = unsafeCastProcessorParametersToCompletelyDifferentType(
+            new ProcessorParameters<>(processorSupplier, name)
+        );
 
-        ProcessorParameters processorParameters = new ProcessorParameters<>(processorSupplier, name);
-        StreamsGraphNode tableNode = new TableProcessorNode<>(name,
-                                                              processorParameters,
-                                                              materializedInternal,
-                                                              null);
+        final StreamsGraphNode tableNode = new TableProcessorNode<>(
+            name,
+            processorParameters,
+            materializedInternal,
+            null
+        );
 
         addGraphNode(tableNode);
 
         builder.internalTopologyBuilder.addProcessor(name, processorSupplier, this.name);
 
         if (shouldMaterialize) {
-            this.builder.internalTopologyBuilder.addStateStore(new KeyValueStoreMaterializer<>(materializedInternal).materialize(), name);
+            this.builder.internalTopologyBuilder.addStateStore(
+                new KeyValueStoreMaterializer<>(materializedInternal).materialize(),
+                name
+            );
```

Comment:
Is our coding style suggesting to always make multi-line function call with more than one parameters? My preference is only use multi-line if there are 3+ parameters, AND if we put them into a single line it would be too long.

Reply:
That was mostly my understanding, except that I thought we would use multi-line if the line was "too long" *regardless* of the number of parameters.

Reply:
I've just reviewed http://kafka.apache.org/coding-guide . It says to use one-param-per-line if the line is longer than 120 chars, which is the case with this code.